### ISSUE TITLE
labels.txt was not found!

### ISSUE THREAD

**Issue by cjltw** [DiKorsch/l1_parts#2]
When I processed the bird data set, I just pointed the path to the original data set, and the labels.txt file was not found at this time. Do I need to pre-process this data set? Looking forward for your response, thank you.

![Screenshot from 2020-08-11 18-08-18](https://user-images.githubusercontent.com/52325368/89885511-c239d400-dbfd-11ea-90e2-b75b94f5ebc4.png)

**Comment by DiKorsch** [comment]
In my case, the CUB labels file has the format 
```
...
<cls_id>
<cls_id>
<cls_id>
<cls_id>
...
```

As far as I know, the original dataset has the format 

```
...
<img_id> <cls_id>
<img_id> <cls_id>
<img_id> <cls_id>
<img_id> <cls_id>
...
```

Hence, you should convert this file accordingly. By the way, the same holds for the `tr_ID.txt`, which is in the original `train_test_split.txt`.

_P.S.: I know, this is some additional work, but unfortunately, it has some weird reasons, why I am using this format. Sorry for the inconvenience._

**Comment by cjltw** [comment]
I'm really sorry to disturb you again. According to your prompt, I generated the labels.txt file and tr_id.txt file. label.txt comes from cls_id in image_class_labels.txt. tr_id is derived from img_id in train_test_split.txt. The content of my CUB_200 folder replacement is shown in the figure below.
![image](https://user-images.githubusercontent.com/52325368/89988926-caa01680-dcb2-11ea-958b-e3f476b4669c.png)
But unfortunately, during the experiment, I used this command `sudo GPU=0 DATASETS=CUB200 ./02_main.sh` and got an error.
![image](https://user-images.githubusercontent.com/52325368/89989248-49954f00-dcb3-11ea-910a-b946a834831b.png)
After researching for a long time, I haven't found what the problem is. I look forward to your suggestions. Thank you very much.

**Comment by DiKorsch** [comment]
First of all, in the `tr_ID.txt` you need to store the split values from `train_test_split.txt` file ( either 0 or 1 ).

Next, your issue is because of the changes in `chainer.iterators.MultiprocessIterator`. It tries to estimate the shared memory size based on the first batch but does this in training mode. As a result, random cropping is used as augmentation, while it should be center cropping for the feature extraction.

The solution to this would be to use a newer `cvdatasets` package (I've added there a hardcoded shared memory size so that the MultiprocessIterator doesn't need to estimate it from the data). Hence, just modify the version number in `requirements.txt` from `cvdatasets~=0.4.2` to `cvdatasets~=0.5.4`. Afterward, rerun the `01_install.sh` script, because the newer package needs to be downloaded into your docker image.

**Comment by cjltw** [comment]
I have tried a lot and failed. I'm really sorry to disturb you again.
When I installed cvdatasets~=0.5.4, I got an error. I tried to install it under python=3.6 and it was successful. However, I tried many ways to modify the python environment in docker, but they all failed.
![image](https://user-images.githubusercontent.com/52325368/90227478-39ae7400-de47-11ea-9ef3-b6ad83f1fa86.png)
I tried a later version of cvdatasets and got another error.
![image](https://user-images.githubusercontent.com/52325368/90227561-5f3b7d80-de47-11ea-8258-4334a6563201.png)
I don't know how to solve this problem, or do I need to upgrade the python version in my docker to 3.6? Looking forward to your reply, thank you.

**Comment by DiKorsch** [comment]
hmm... this is again some versioning problem... Chainer4.2 was using python3.5 and the f-strings were introduced in python3.6. I've just updated the master branch and created the `stable` tag. Please check out this one and try it out. 

_Don't forget to clean your docker images and containers with `docker container prune` and `docker image prune`. Otherwise, you will have old images and containers laying around and taking away the HD space_

**Comment by cjltw** [comment]
Here, I have an error, you may need to give the mirror a tag. 
![image](https://user-images.githubusercontent.com/52325368/90328910-ea954a00-dfd2-11ea-84a7-d4156b9fa91c.png)

**Comment by cjltw** [comment]
I am very happy that I finally set up the environment, thank you for your recent guidance. Unfortunately, during the operation, I got another error, but I didn't find a problem with the settings. At first I thought that the labels.txt file was not processed correctly, but the problem seemed to be more serious when I changed the labels to 0~199. Looking forward to your reply, thank you.
![image](https://user-images.githubusercontent.com/52325368/90331922-00af0480-dfeb-11ea-89c9-283249fca62d.png)

**Comment by DiKorsch** [comment]
leave the labels as they are, so 1-200. To handle this, I created the `label_shift` [option](https://github.com/DiKorsch/l1_parts/blob/master/pipeline.sh#L54). This is handled later in the different steps of the pipeline.

**Comment by cjltw** [comment]
Then why do I get the above error when I use 1-200?

**Comment by DiKorsch** [comment]
oh, ok, I misunderstood your message. Maybe the label shift is not applied somewhere, I will need to check this. Unfortunately, I will be able to do this only on Tuesday.

**Comment by cjltw** [comment]
Thank you very much for your help. During this time, I can study your paper and think about how to combine your methods with our work. I think it will be fun.

**Comment by DiKorsch** [comment]
You may also contact me directly via email, if you like.
